src/main/scala/com/mwz/sonar/scala/scoverage/ScoverageSensor.scala

```diff
@@ -0,0 +1,171 @@
+              for ((linenum, hits) <- fileCoverage.linesCoverage) {
+                coverage.lineHits(linenum, hits)
+              }
+              coverage.save()
+            }
+            case None => {
+              logger.warn(
+                s"[scoverage] The file: '${filename}', has no scoverage information associated to it."
+              )
+            }
+          }
+        }
+      }
+      case Failure(ex) => {
+        logger.error(
+          s"""[scoverage] Aborting the scoverage sensor execution,
+             |cause: an error occurred while reading the scoverage report file: '${reportFilename}',
+             |the error was: ${ex.getMessage}.""".stripMargin
+        )
+      }
+    }
+  }
+
+  /** Returns all scala main files from this module */
+  private[this] def getModuleSourceFiles(fs: FileSystem): Iterable[InputFile] = {
+    val predicates = fs.predicates
+    val predicate = predicates.and(predicates.hasLanguage(Scala.Key), predicates.hasType(InputFile.Type.MAIN))
+    fs.inputFiles(predicate).asScala
+  }
+
+  /** Returns the module base path */
+  private[this] def getModuleBaseDirectory(fs: FileSystem): String = {
+    val moduleAbsolutePath = Paths.get(fs.baseDir().getAbsolutePath).normalize()
+    val currentWorkdirAbsolutePath = Paths.get("./").toAbsolutePath().normalize()
+    val moduleBasePath = currentWorkdirAbsolutePath.relativize(moduleAbsolutePath).toString
+    PathUtils.sanitizePath(moduleBasePath)
+  }
+
+  /** Returns the filename of the scoverage report for this module */
+  private[this] def getScoverageReportFilename(settings: Configuration): String = {
+    settings
+      .get(ScoverageSensorInternal.ScoverageReportPathPropertyKey)
+      .toOption
+      .getOrElse(
+        ScoverageSensorInternal.getDefaultScoverageReportPath(Scala.getScalaVersion(settings))
+      )
+  }
+
+  /** Saves the [[ScoverageMetrics]] of a component */
+  private[this] def saveComponentScoverage(
+    context: SensorContext,
+    component: InputComponent,
+    scoverage: Scoverage
+  ): Unit = {
+    context
+      .newMeasure[java.lang.Integer]()
+      .on(component)
+      .forMetric(ScoverageMetrics.totalStatements)
+      .withValue(scoverage.totalStatements)
+      .save()
+
+    context
+      .newMeasure[java.lang.Integer]()
+      .on(component)
+      .forMetric(ScoverageMetrics.coveredStatements)
+      .withValue(scoverage.coveredStatements)
+      .save()
+
+    context
+      .newMeasure[java.lang.Double]()
+      .on(component)
+      .forMetric(ScoverageMetrics.statementCoverage)
+      .withValue(scoverage.statementCoverage)
+      .save()
+
+    context
+      .newMeasure[java.lang.Double]()
+      .on(component)
+      .forMetric(ScoverageMetrics.branchCoverage)
+      .withValue(scoverage.branchCoverage)
+      .save()
+  }
+}
+
+object ScoverageSensorInternal {
+  private val SensorName = "Scoverage Sensor"
+  private val ScoverageReportPathPropertyKey = "sonar.scala.scoverage.reportPath"
```

Comment:
It is a good idea to align the key name with the scapegoat plugin and use `sonar.scala` prefix, but this would be a breaking change :/ so perhaps we should wait with making it until we release version 7 of this plugin for SonarQube 7?

Another option would be to deprecate the `sonar.scoverage.reportPath` key and add a warning and have support for both keys until we remove it in version 7.  I'm aware that this option would require more work so I'd be happy to keep the old key for now and we can deprecate it and add the support for both keys later.

Reply:
I agree that the change should be made with more caution.

Probably the best option is to stay with the old key for now.